### ISSUE TITLE
How to publish all available versions of ansible to docker hub

### ISSUE THREAD

**Issue by GabLeRoux** [GabLeRoux/ansible-docker-image#2]
I'm not sure what would be the best way to do so, but right now I'm using [automated build as described here](https://docs.docker.com/docker-hub/builds/).

I think it would be possible to create a commit foreach versions by changing the ENV for the ansible version, but that would be lame and there must be a better way. I also want the builds to be transparent and accessible through the docker hub.

Using travis could be a good workaround, but it makes the registry lose the github link and it's sad. I'm open to ideas here unless I find a solution first :P

**Comment by GabLeRoux** [comment]
```bash
pip install ansible==
```

```
(from versions: 1.0, 1.1, 1.2, 1.2.1, 1.2.2, 1.2.3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4, 1.4, 1.4.1, 1.4.2, 1.4.3, 1.4.4, 1.4.5, 1.5, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.5.5, 1.6, 1.6.1, 1.6.2, 1.6.3, 1.6.4, 1.6.5, 1.6.6, 1.6.7, 1.6.8, 1.6.9, 1.6.10, 1.7, 1.7.1, 1.7.2, 1.8, 1.8.1, 1.8.2, 1.8.3, 1.8.4, 1.9.0.1, 1.9.1, 1.9.2, 1.9.3, 1.9.4, 1.9.5, 1.9.6, 2.0.0.0, 2.0.0.1, 2.0.0.2, 2.0.1.0, 2.0.2.0, 2.1.0.0, 2.1.1.0, 2.1.2.0, 2.1.3.0, 2.1.4.0, 2.1.5.0, 2.1.6.0, 2.2.0.0, 2.2.1.0, 2.2.2.0, 2.2.3.0, 2.3.0.0, 2.3.1.0, 2.3.2.0, 2.3.3.0, 2.4.0.0, 2.4.1.0, 2.4.2.0, 2.4.3.0, 2.4.4.0, 2.4.5.0, 2.4.6.0, 2.5.0a1, 2.5.0b1, 2.5.0b2, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.5.2, 2.5.3, 2.5.4, 2.5.5, 2.5.6, 2.5.7, 2.5.8, 2.5.9, 2.5.10, 2.5.11, 2.5.12, 2.5.13, 2.5.14, 2.6.0a1, 2.6.0a2, 2.6.0rc1, 2.6.0rc2, 2.6.0rc3, 2.6.0rc4, 2.6.0rc5, 2.6.0, 2.6.1, 2.6.2, 2.6.3, 2.6.4, 2.6.5, 2.6.6, 2.6.7, 2.6.8, 2.6.9, 2.6.10, 2.6.11, 2.6.12, 2.6.13, 2.7.0.dev0, 2.7.0a1, 2.7.0b1, 2.7.0rc1, 2.7.0rc2, 2.7.0rc3, 2.7.0rc4, 2.7.0, 2.7.1, 2.7.2, 2.7.3, 2.7.4, 2.7.5, 2.7.6, 2.7.7)
```

So these are all the available versions :)

Here's a one version per line list:

```
1.0
1.1
1.2
1.2.1
1.2.2
1.2.3
1.3.0
1.3.1
1.3.2
1.3.3
1.3.4
1.4
1.4.1
1.4.2
1.4.3
1.4.4
1.4.5
1.5
1.5.1
1.5.2
1.5.3
1.5.4
1.5.5
1.6
1.6.1
1.6.2
1.6.3
1.6.4
1.6.5
1.6.6
1.6.7
1.6.8
1.6.9
1.6.10
1.7
1.7.1
1.7.2
1.8
1.8.1
1.8.2
1.8.3
1.8.4
1.9.0.1
1.9.1
1.9.2
1.9.3
1.9.4
1.9.5
1.9.6
2.0.0.0
2.0.0.1
2.0.0.2
2.0.1.0
2.0.2.0
2.1.0.0
2.1.1.0
2.1.2.0
2.1.3.0
2.1.4.0
2.1.5.0
2.1.6.0
2.2.0.0
2.2.1.0
2.2.2.0
2.2.3.0
2.3.0.0
2.3.1.0
2.3.2.0
2.3.3.0
2.4.0.0
2.4.1.0
2.4.2.0
2.4.3.0
2.4.4.0
2.4.5.0
2.4.6.0
2.5.0a1
2.5.0b1
2.5.0b2
2.5.0rc1
2.5.0rc2
2.5.0rc3
2.5.0
2.5.1
2.5.2
2.5.3
2.5.4
2.5.5
2.5.6
2.5.7
2.5.8
2.5.9
2.5.10
2.5.11
2.5.12
2.5.13
2.5.14
2.6.0a1
2.6.0a2
2.6.0rc1
2.6.0rc2
2.6.0rc3
2.6.0rc4
2.6.0rc5
2.6.0
2.6.1
2.6.2
2.6.3
2.6.4
2.6.5
2.6.6
2.6.7
2.6.8
2.6.9
2.6.10
2.6.11
2.6.12
2.6.13
2.7.0.dev0
2.7.0a1
2.7.0b1
2.7.0rc1
2.7.0rc2
2.7.0rc3
2.7.0rc4
2.7.0
2.7.1
2.7.2
2.7.3
2.7.4
2.7.5
2.7.6
2.7.7
```

I put that in a `versions.txt` and I ran the following bash script:

```bash
cat versions.txt | xargs -I {} bash -c "echo 'ansible=={}' > requirements.txt && git add requirements.txt && git commit -m 'Set version to {}' && git tag {} && git push && git push --tags"
```

Above command set update the `requirements.txt`, pushes commit and tag on each step to trigger docker hub build and that's it 🎉

Note: I've found a few limitations with this. Docker hub build history page lacks pagination so there may have some builds that failed.

I did encounter following error before doing all of the above:

```
Downloading https://files.pythonhosted.org/packages/68/9e/49196946aee219aead1290e00d1e7fdeab8567783e83e1b9ab5585e6206a/pycparser-2.19.tar.gz (158kB)
Exception:
Traceback (most recent call last):
File "/usr/local/lib/python3.7/site-packages/pip/_internal/cli/base_command.py", line 176, in main
status = self.run(options, args)
File "/usr/local/lib/python3.7/site-packages/pip/_internal/commands/install.py", line 346, in run
session=session, autobuilding=True
File "/usr/local/lib/python3.7/site-packages/pip/_internal/wheel.py", line 886, in build
assert have_directory_for_build
AssertionError
You are using pip version 19.0.1, however version 19.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
Removing intermediate container 3bee7ae28d7c
The command '/bin/sh -c apk add --update --no-cache --virtual .build-deps make gcc python3-dev && pip install --no-cache-dir -r requirements.txt && apk del .build-deps' returned a non-zero code: 2
```

This was fixed by upgrading pip in 9ef0678

docker hub is still building all of them right now, but it seems to work fine.

Versions bellow 1.8 seems to fail building, that's probably something to do with python version. This could go in a different issue, but I think this should cover a lot of use case. 🎉

**Comment by GabLeRoux** [comment]
Closing as following script works fine:  
https://github.com/GabLeRoux/ansible-docker-image/blob/master/scripts/batch_update_versions.sh
